### PR TITLE
Async load gridicons

### PR DESCRIPTION
**Pull request by flootr**
• 61eae9a creates a separate chunk for each gridicon
• f582517 creates a separate `vendor-gridicons` chunk which gathers imported gridicons
• bef8e90 creates a separate `gridicons~build` chunk which includes the whole gridicon module

### CODE REVIEW

**Comment by dmsnell** [comment]
Since it came up, how would this deal with modules that don't exist, such as when we try to load a slug that isn't in the library?

Further, this ends up loading our SVGs as JavaScript files and I wonder if that's a negative implication? We can't see the SVGs load in the developer tools as images anymore and we can't preview them. It appears to send the object structure so even the SVG text is lost in here.

**Comment by flootr** [comment]
> Since it came up, how would this deal with modules that don't exist, such as when we try to load a slug that isn't in the library?

We probably need to provide a fallback. In case an icon doesn't exist Webpack should throw an error which we should be able to `.catch()`.

> Further, this ends up loading our SVGs as JavaScript files and I wonder if that's a negative implication? 

This is no different than the current solution, isn't it? I agree that loading SVGs directly would be a preferable solution. In the end this PR is only an experiment.

File: client/components/async-gridicons/index.jsx [hunk 1]
```diff
@@ -9,14 +9,18 @@ import React from 'react';
  * Internal dependencies
  */
 import AsyncLoad from 'components/async-load';
+import FallbackIcon from './fallback';
 
 function AsyncGridicon( { icon = '', ...rest } ) {
 	return (
```
Comment (flootr) [inline client/components/async-gridicons/index.jsx]: This is marked red, not sure why though 🤔 

screen shot 2018-09-24 at 18 01 22

File: client/components/async-gridicons/index.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import AsyncLoad from 'components/async-load';
+import FallbackIcon from './fallback';
+
+function AsyncGridicon( { icon = '', ...rest } ) {
+	return (
```
Comment (flootr) [inline client/components/async-gridicons/index.jsx]: This is marked red, not sure why though 🤔 

screen shot 2018-09-24 at 18 01 22

File: client/components/async-gridicons/index.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ function AsyncGridicon( { icon = '', ...rest } ) {
 						callback( FallbackIcon );
 					} );
 			} }
-			placeholder={ null }
+			placeholder={ <FallbackIcon /> }
```
Comment (flootr) [inline client/components/async-gridicons/index.jsx]: Ha! Good catch @blowery 😄

**Comment by blowery** [comment]
> creates a separate vendor-gridicons chunk which gathers imported gridicons

how big was that chunk?

**Comment by blowery** [comment]
@adlk another odd desktop build failure...

**Comment by blowery** [comment]
ICFY report http://iscalypsofastyet.com/branch?branch=try/async-gridicon

vendors~build down 20k zipped, build up 3k(?), manifest up 4.8k, net savings of 12k. 

Going to try again with all the gridicons in one chunk. Might mitigate the manifest and build increases.

**Comment by flootr** [comment]
@blowery in general, yes.

> vendors~build down 20k zipped, build up 3k(?), manifest up 4.8k, net savings of 12k.

Do you know where the 3k up comes from?

> OK, gridicons in one chunk: 2.6k increase in build, 32 byte increase in manifest, same 20K drop in vendors~build. So a 18k drop instead of 12k. I'll take that.

Do we need to take into account how often a user needs to download that gridicon chunk? Caching is probably more efficient using single chunks but gridicons also don't change often so it should be okay.

**Comment by blowery** [comment]
No idea where the 3k was coming from.

> Do we need to take into account how often a user needs to download that gridicon chunk? Caching is probably more efficient using single chunks but gridicons also don't change often so it should be okay.

I think it'll be ok. Gridicons change very rarely.

**Comment by blowery** [comment]
I did some extra investigation to exclude examples and index, but it seems they're naturally excluded? I'm not sure why that is, but it appears that way.

I think this is good to go.

**Review by blowery — APPROVED** [review]

**Comment by sirreal** [comment]
After revert, a second attempt was landed #27527

**Pull request merged**